gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -149,7 +150,11 @@ void initialize(EditObservationFragmentArgs args) {
 
   Optional<Response> getSavedOrOriginalResponse(String fieldId) {
     if (responses.isEmpty()) {
-      return originalObservation.getResponses().getResponse(fieldId);
+      if (originalObservation == null){
```

Comment:
Alternatively:
`return Optional.fromNullable(originalObservation).map(o -> o.getResponses().getResponse(fieldId));`

Reply:
I don't think this works because when `originalObservation` is null an `Optional<Observation>` would be returned rather than the required `Optional<Response>`.

Reply:
Out of curiosity, have you tried and it doesn't compile? When `originalObservation` is null, Optional.fromNullable(originalObservation) will be Optional.empty(). I think the generic closure is inferred from the type returned by the lambda in `map()`.

Reply:
I tried it and it doesn't compile. 

![image](https://user-images.githubusercontent.com/873212/98373380-11de3f00-2037-11eb-9ff0-1cd6749c1a91.png)

Reply:
Ok, thanks for checking!